test/e2e_federation/util.go

```diff
@@ -141,21 +141,84 @@ func createService(clientset *fedclientset.Clientset, namespace, name string) (*
 	}
 
 	By(fmt.Sprintf("Trying to create service %q in namespace %q", service.Name, namespace))
-	return clientset.Services(namespace).Create(service)
+	var err error
```

Comment:
Given that we have 2 functions now `createService` and `createLBService`, checking and deleting if service shard is created in all member cluster in case of simple service is a overkill. I think the logic to check successful creation of service shard and retrying should be good enough in the scenario of `createLBService`.